### PR TITLE
allow access to dapp without connected wallet

### PR DESCRIPTION
**Pull request by rolandpo**
issue #1007

- swap/wrap forms show even if no wallet is connected
- contract table on swap page also shows, and assumes no allowances have been granted
- swap button greyed out and display 'Connect Wallet' message
- coin balances are hidden
- transaction history page stays the same as before

### CODE REVIEW

**Comment by micahalcorn** [comment]
Thanks @rolandpo 👍 Any chance we could get a screenshot of each state or a staging deployment of this branch?

**Comment by rolandpo** [comment]
Here's a screenshot of the swap page without an account. Swap button is disabled and prompt to connect wallet, balance header fields are empty and token balances are hidden

Screenshot 2022-06-02 at 11 33 16

**Comment by sparrowDom** [comment]
@rolandpo trying to test this functionality. I wasn't able to connect Metamask to the dapp.

**Comment by rolandpo** [comment]
interesting, I'm not having that problem

**Comment by sparrowDom** [comment]
I got that on Android using Chrome. What environment do you have?

**Comment by rolandpo** [comment]
ok I was just talking desktop. On mobile clicking the MetaMask button gives me a 'No Ethereum wallet detected' message, but going through the MM app browser it connects fine. Which probably isn't ideal but it works the same way as the regular deployed dapp does from what I'm seeing

**Comment by sparrowDom** [comment]
Ahh sorry wasn't paying attention and though this was another PR. Yes I was testing this on Desktop. Let me try again.

**Comment by sparrowDom** [comment]
Must have been something wrong with my environment. I can not reproduce the issue. Can you make these 2 additions: 
- do not make it possible to click on swap route when wallet is not connected. Or instead of a modal show a toast saying "Connect a wallet first"
- `Swap` navigation item should be visible when wallet is not connected:
Screenshot 2022-06-17 at 15 09 18

**Comment by rolandpo** [comment]
I disabled contract selection when no account is connected. On the weird navbar stuff I remember seeing that before, but can't reproduce now. Let me know if it's still an issue

File: dapp/src/components/buySell/SwapHomepage.js [hunk 1]
```diff
@@ -39,12 +39,17 @@ const SwapHomepage = ({
 
   // mint / redeem
   const [swapMode, setSwapMode] = useState(
-    localStorage.getItem(lastSelectedSwapModeKey) || 'mint'
+    typeof window !== 'undefined'
```
Comment (sparrowDom) [inline dapp/src/components/buySell/SwapHomepage.js:42]: using `process.browser` is a bit more reliable then checking for presence of `window`

File: dapp/src/components/buySell/SwapHomepage.js [hunk 2]
```diff
@@ -39,12 +39,17 @@ const SwapHomepage = ({
...
+      ? localStorage.getItem(lastSelectedSwapModeKey)
+      : 'mint'
   )
   const previousSwapMode = usePrevious(swapMode)
   const [buyErrorToDisplay, setBuyErrorToDisplay] = useState(false)
 
-  const storedSelectedCoin = localStorage.getItem(lastUserSelectedCoinKey)
+  const storedSelectedCoin =
+    typeof window !== 'undefined'
```
Comment (sparrowDom) [inline dapp/src/components/buySell/SwapHomepage.js:50]: `process.browser` here as well

File: dapp/src/components/buySell/SwapHomepage.js [hunk 3]
```diff
@@ -114,7 +119,10 @@ const SwapHomepage = ({
   )
 
   useEffect(() => {
-    let lastUserSelectedCoin = localStorage.getItem(lastUserSelectedCoinKey)
+    let lastUserSelectedCoin =
+      typeof window !== 'undefined'
```
Comment (sparrowDom) [inline dapp/src/components/buySell/SwapHomepage.js:123]: and here

File: dapp/src/components/buySell/SwapHomepage.js [hunk 4]
```diff
@@ -316,75 +324,81 @@ const SwapHomepage = ({
...
-          position: relative;
-        }
-
-        @media (max-width: 799px) {
-          .swap-homepage {
-            padding: 23px 20px 20px 20px;
-          }
-        }
-      `}</style>
+      {typeof window !== 'undefined' && (
```
Comment (sparrowDom) [inline dapp/src/components/buySell/SwapHomepage.js:327]: here too

File: dapp/src/components/wrap/WrapHomepage.js [hunk 5]
```diff
@@ -40,7 +40,9 @@ const WrapHomepage = ({
 }) => {
   // mint / redeem
   const [swapMode, setSwapMode] = useState(
-    localStorage.getItem(lastSelectedSwapModeKey) || 'mint'
+    typeof window !== 'undefined'
```
Comment (sparrowDom) [inline dapp/src/components/wrap/WrapHomepage.js:43]: and here

File: dapp/src/components/wrap/WrapHomepage.js [hunk 6]
```diff
@@ -251,59 +253,63 @@ const WrapHomepage = ({
...
-        .wrap-homepage {
-          margin: 0px -1px -1px -1px;
-          border: solid 1px #cdd7e0;
-          border-radius: 10px;
-          background-color: #fafbfc;
-          min-height: 350px;
-          padding: 35px 40px 40px 40px;
-          position: relative;
-        }
+      {typeof window !== 'undefined' && (
```
Comment (sparrowDom) [inline dapp/src/components/wrap/WrapHomepage.js:256]: ...

**Review by sparrowDom — APPROVED** [review]
The navbar element not showing was contingent to some environmental variables... that logic was no longer relevant so have removed that. 

Left 1 comment inline (well a couple of places but in effect it is the same type of comment). Amazing job on this feature, can't wait to get this deployed. 💪

**Comment by sparrowDom** [comment]
@rolandpo actually it is such a minor change, that I wen't ahead and did it. Thanks again amazing work 👏
